backends/p4tools/modules/testgen/core/small_step/extern_stepper.cpp

```diff
@@ -37,7 +37,9 @@ std::vector<std::pair<IR::StateVariable, const IR::Expression *>> ExprStepper::s
     ExecutionState &nextState, const std::vector<IR::StateVariable> &flatFields,
     int varBitFieldSize) {
     std::vector<std::pair<IR::StateVariable, const IR::Expression *>> fields;
-    for (const auto &fieldRef : flatFields) {
+    // Make a copy of the StateVariable so it can be modified in the varbit case (and it is just a
```

Comment:
Hmm, in this case we are making a copy for the common case (no Varbit). Is that really better?

Reply:
This is quite small class (one pointer + vptr) so I'd say this does not matter either way. Could be even better since there will be no dereference, but could be also exactly the same if the optimizer is smart enough (I haven't checked).

Reply:
Yeah I think that is a check we should keep in this particular setter. The other code can be moved out.

Reply:
I'm confused, you have presubmably replied to a different thread, did you mean that the check for type being set in the expression should be (also) in `SymbolicEnv::set`?

Reply:
Yes, the comment was meant for https://github.com/p4lang/p4c/pull/4300#discussion_r1442624167. Not sure what happened.